go/Internal/commands/deleteSongCommand.go

```diff
@@ -0,0 +1,73 @@
+package internal
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Canhassi12/scylla-cloud-getting-started/Internal/database"
+	"github.com/scylladb/gocqlx/v2"
+)
+
+type DeleteSongCommandInterface interface {
+	Delete(*gocqlx.Session) error
+}
+
+type DeleteSongCommand struct {
+
+}
+
+func NewDeleteSongCommand() *DeleteSongCommand {
+	return &DeleteSongCommand{}
+}
+
+func (cmd *DeleteSongCommand) Delete(session *gocqlx.Session) error {
+	songs, err := NewListSongCommand().List(session); if err != nil {
+		return err
+	}
+
+	index, err := cmd.selectSongToDelete(songs); if err != nil {
+		return err
+	}
+
+	if index >= 0 && index < len(songs) {
+		songToDelete := songs[index]
+
+		q := session.Query(`DELETE FROM media_player.playlist WHERE id = ?`,
+		[]string{":id"}).
+		BindMap(map[string]interface{} {
+			":id": songToDelete.Id,
+		})
+
+		err := q.Exec(); if err != nil {
+			return fmt.Errorf("error to exec delete query %w", err)
+		}
+	}
+		
+	return nil
+}
+
+func (cmd *DeleteSongCommand) selectSongToDelete(songs []database.Song) (int, error) {
+	for i, song := range songs {
+		fmt.Printf("Index: %d  | Song: %s | Album: %s | Artist: %s | Created At: %s\n", i+1, song.Title, song.Album, song.Artist, song.Created_at)
```

Comment:
Is there any particular reason to not use `song.String()` here?

Reply:
its because it doesn't need